src/GraphQL/Types/GraphQLTypeReference.cs

```diff
@@ -9,7 +9,6 @@ public class GraphQLTypeReference : InterfaceGraphType, IObjectGraphType
     {
         public GraphQLTypeReference(string typeName)
         {
-            Name = "__GraphQLTypeReference";
```

Comment:
This name does not meet the specification.  `__` is reserved for fixed introspection type list so changed it to be `GraphQLTypeReference`.